ui/src/components/admin/Model/ModelSingleHeader.js

```diff
@@ -100,20 +101,19 @@ const ModelSingleHeader = ({ modelName }) => {
   useEffect(() => {
     const refreshModelData = async () => await fetchModelData(modelName);
 
-    if (modelName && isPublishingModel(modelName) && (!publishState || !publishState.current)) {
+    if (modelName && isPublishingModel(modelName)) {
       publishState.current = 'publishing';
     }
 
     if (
       modelName &&
-      publishState.current === 'publishing' &&
+      (publishState.current === 'publishing' || !publishState.current) &&
```

Comment:
Adding a check for `!publishState.current` since some publishes are completing faster than the time it takes to set the state to `'publishing'` (i.e. by the time this useEffect is called, the publish has already finished. We still need to refresh the model data in this case, otherwise the page is left with a "Changes Not Yet Published" label)

By also allowing `!publishState.current`, we ensure that we refresh the model data even when the publish happens too fast.